### PR TITLE
Win symlinks

### PR DESCRIPTION
**Pull request by bogdanm**
This PR changes the 'yotta link' implementation from junctions to
symbolic links, which has a number of advantages:

- symbolic links can point to pretty much anything (including network paths)
- it seems to be harder to mess things up using symbolic links. Junctions
can behave weirdly when they're deleted (read: in some circumstances,
deleting a junction will also delete whatever directory the junction is
pointing to. Happened to me and others a number of times). Directory symbolic
links can safely be deleted using "rmdir" (the implementation of 'rmF' was
changes in fsutils_win.py to reflect this)

Also, "yotta unlink" works in Windows now.
Tested on Win7 64 bit, NTFS file system.

### CODE REVIEW

**Comment by 0xc0170** [comment]
I lost couple of repositories with staged changes doing unlinking :-)  
:dancer: unlink for windows will be back !

**Comment by bogdanm** [comment]
For me, `yt unlink` never worked under Windows, failing with an `access denied` error or something similar to that. However, when I wanted to "fix" that and tried to remove the linked module by hand from the command line, it'd sometimes (not all the time!) delete both the junction **and** the directory it was pointing to. Fun fun fun.

**Comment by bogdanm** [comment]
Do not merge yet, I believe I can make it less ugly.

**Comment by autopulated** [comment]
ack

File: setup.py [hunk 1]
```diff
@@ -10,9 +10,9 @@
 def read(fname):
     return open(os.path.join(os.path.dirname(__file__), fname)).read()
 
-# we need 'ntfsutils' in Windows
+# we need 'pywin32' in Windows
 if os.name == 'nt':
-    platform_deps = ['ntfsutils>=0.1.3,<0.2']
+    platform_deps = ['pywin32']
```
Comment (autopulated) [inline setup.py]: please add a version specification here

**Comment by autopulated** [comment]
It seems like someone should have encapsulated this into a separate python module – maybe we should do that?

File: setup.py (hunk 1 above)
Comment (bogdanm) [inline setup.py]:
```
> pip list
[snip]
pywin32 (219)
[snip]
```

Good luck with that :)

**Comment by bogdanm** [comment]
OK, so apparently I was wrong. Symlinks don't have a "better" behaviour when it comes to deleting stuff. I did a little test which involved creating three types of links (symbolic link, hard link, junction) to a directory, then trying to delete the link the Windows way (`rd /s /q <linkname>`) and the Linux way (`rm -rf <linkname>`, using rm from https://github.com/bmatzelle/gow/ (IIRC)). Results:

|              | rm -rf | rd /s /q |
|-------------|---------|----------|
| Symlink   | **Deletes everything** | Deletes just the link |
| Hard link | Access denied | Access denied |
| Junction  | **Deletes everything** | Deletes just the link |

I was unable to create any hard links at all (`Access is denied`), but other than that, it looks like my problem with deleted files came from my attempt to use `rm` instead of `rd` (it's concievable that the Windows port of `rm` doesn't handle links very well, since the link semantics are different in Windows and Linux). With that in mind, the only thing this PR really fixes for me is `yotta unlink`, which can probably be fixed easier in the original junction-based implementation.
That said, did other people experience this problem? @0xc0170, do you remember how you managed to get your repositories deleted? Again, `yotta unlink` didn't delete anything for me in Windows, because it never worked; everytime I used `yotta unlink` I got an `Access is denied` exception.

**Comment by 0xc0170** [comment]
I'll have to spend some time to do this - I am using a commander based on git bash. I have tried a minute ago to link local repo, and once I did yt link, the repo is gone. This is the console output (yt link within cmsis-core-k64f is not shown as it was in another console tab).

```
C:\Code\git_repo\github\yotta_thing\ksdk-a
λ yt link cmsis-core-k64f
info: C:\Code\git_repo\github\yotta_thing\ksdk-a\yotta_modules\cmsis-core-k64f -> C:\Python27\Lib\yotta_modules\cmsis-core-k64f -> c:\Code\git_repo\github\yotta_thing\cmsis-core-k64f

C:\Code\git_repo\github\yotta_thing\ksdk-a
λ yt ls
mbed-hal-ksdk-mcu 0.1.4
|_ mbed-hal 0.2.0
| |_ mbed 3.0.2 yotta_modules\mbed
| | \_ cmsis-core 0.1.2 yotta_modules\cmsis-core
| |   \_ cmsis-core-freescale 0.1.1 yotta_modules\cmsis-core-freescale
| |     \_ cmsis-core-k64f ~0.1.0 missing
| \_ mbed-hal-freescale 0.1.4 yotta_modules\mbed-hal-freescale
\_ mbed-hal-k64f 0.1.7
  |_ uvisor-lib 0.6.4 yotta_modules\uvisor-lib
  \_ mbed-hal-frdm-k64f 0.1.3 yotta_modules\mbed-hal-frdm-k64f


C:\Code\git_repo\github\yotta_thing\ksdk-a
λ yt build
info: get versions for cmsis-core-k64f
info: download cmsis-core-k64f@0.1.1 from https://yotta-private.herokuapp.com
```
Once I did yt link, the repo is gone and reported as missing. I'll find more time next week to find exact commands when this happens on my machine with the logging, so I can share as above which was just a quick check.

**Comment by bogdanm** [comment]
Let me guess - that repo was linked before? Cause I also found the perfect way to reproduce this:

- link a module in your current module
- `yt ls` in the current module will show you the linked module propely
- linked the same module again
- `yt ls` will show `missing` and the directory of the module being linked in is gone :sob:

I've figured out the cause too. When linking, `yotta` will first try to `rmRf` the directory that's being linked. On POSIX, this ends up calling `unlink`, which is the right thing to do. On Windows, it goes ahead and deletes everything :facepunch: So I'm gonna go ahead and create a patch for this issue first.
Of course, this means that this PR shouldn't be merged. Sorry for the noise.

**Pull request closed without merging**